### PR TITLE
[dhcp6pd] add basic DHCPv6-PD support

### PR DESCRIPTION
**Pull request by erjiaqing**
This PR includes change for build flags and feature flags.

### CODE REVIEW

**Review by superwhd — APPROVED** [review]
LGTM 👍

**Comment by erjiaqing** [comment]
@jwhui PTAL, thanks.

File: third_party/openthread/CMakeLists.txt [hunk 1]
```diff
@@ -67,6 +67,7 @@ set(OT_TCP OFF CACHE STRING "disable TCP" FORCE)
 set(OT_TREL ${OTBR_TREL} CACHE STRING "enable TREL" FORCE)
 set(OT_UDP_FORWARD OFF CACHE STRING "disable udp forward" FORCE)
 set(OT_UPTIME ON CACHE STRING "enable uptime" FORCE)
+set(OT_BORDER_ROUTING_DHCP6_PD ${OTBR_DHCP6_PD} CACHE STRING "enable dhcpv6 pd support in border routing" FORCE)
```
Comment (jwhui) [inline third_party/openthread/CMakeLists.txt:70]: nit: maintain alphabetical order

Comment (erjiaqing) [reply third_party/openthread/CMakeLists.txt]: done

**Review by jwhui — APPROVED** [review]

**Pull request merged**